### PR TITLE
[examples] summarization/bart/finetune.py supports t5

### LINKED ISSUE #3576: T5 fine tune for seq2seq generation
# ❓ Questions & Help

## Details

Hi
Is a script available for fine-tuning T5 base or large to to do seq2seq generative tasks like translation or dialog generation?

https://github.com/huggingface/transformers/blob/master/examples/run_generation.py

Doesn't seem to have T5

### LINKED ISSUE #3813: T5 prediction using fine-tuned model
After fine-tuning the T5 model on my own dataset, when I use the fine-tuned model to predict for test set using the following command:

python '/content/transformers-master/examples/summarization/bart/run_bart_sum.py' --data_dir='/content/drive/My Drive/two_keywords/' --model_type=t5 --output_dir=/content/t5 --do_predict --model_name_or_path=t5-small
Error generated: 
Screenshot 2020-04-13 at 6 18 47 PM

# ❓ Questions & Help

## Details

**A link to original question on Stack Overflow**:

### PR DESCRIPTION
**Pull request by sshleifer**
- we were passing attention_mask as an arg, not a kwarg, causing `test_step` to break.
- That case is now covered in the unittest, unittests also cover the t5 model.
- renamed run_bart_sum.py to finetune.py since it is model agnostic. 
- The `bart/` and `t5` subdirectories should be consolidated in a future PR.

This took 15 mins because of underlying infrastructure: unittests for examples and tiny models on S3 :)

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Super! LGTM

**Review by patrickvonplaten — APPROVED** [review]

**Comment by prabalbansal** [comment]
Thanks @sshleifer for the quick fix. Just a small query where it will save the output sentences. A file is generated in output_dir with only losses specified when --do_predict is passed in Argument.
What if I want to generate for unknown inputs using fine-tuned model.
